### ISSUE TITLE
Controllers don't show up on Oculus Quest when VIU_OCULUSVR_AVATARS is false

### ISSUE THREAD

**Issue by unoctium1** [ViveSoftware/ViveInputUtility-Unity#219]
Tested on VIU 12.2 and both Unity 2019.4 and Unity 2021.1, and replicated with both standalone (via Oculus Link) and android. 

It seems like the issue is with VIUOculusVRRenderModel.cs as the SetDeviceIndex method doesn't do anything do to the conditionals. I was able to fix this locally by just wrapping RenderModelCreator in OculusVRModule in an #if VIU_OCULUSVR_AVATAR block, so that it falls back to the default RenderModelCreator, but this probably isn't the most graceful solution

**Comment by chengnay** [comment]
@unoctium1 What is your testing environment setup?
Which Unity/Plugin versions did you import/install?
And this solution is only for Oculus Quest? What about Quest 2? or other Oculus devices?

**Comment by chengnay** [comment]
@unoctium1 I tested your solution, and I found out that there is default render model(VIVE controller), but there is no input. Are you also getting the same result? Or from your side, you can have button inputs?

We will need some time to fix this issue for using latest Oculus Integration SDK.
Sorry for the inconvenience.

**Comment by unoctium1** [comment]
@chengnay No, I'm seeing Oculus Quest controllers and receiving button input. However, what you described (Vive controller & no button input) happens to me with or without my changes if the Oculus SDK is not being used and Unity falls back to OpenVR. Is that maybe happening on your end?

**Comment by chengnay** [comment]
@unoctium1 Which version of Oculus XR Plugin did you install? v1.9.1?

**Comment by chengnay** [comment]
@unoctium1 
![image](https://user-images.githubusercontent.com/33050705/118070916-5e854700-b3d9-11eb-8f35-3e61d8bb0e92.png)

![image](https://user-images.githubusercontent.com/33050705/118071439-5a0d5e00-b3da-11eb-80be-895f0ca74de2.png)

On my side, it reads as Oculus_Link_Quest_2 when I plug in Oculus Quest.
Anyway, this issue will be fixed in next update.
Thanks for using VIU. :)

**Comment by chengnay** [comment]
@unoctium1 I think I know the root cause to the VIU_OCULUSVR_AVATARS is not listed in the defined symbol field.

Please try this [batch file](https://github.com/ViveSoftware/ViveInputUtility-Unity/blob/develop/DeleteAllAsmdef.bat) to remove the asmdef files in VIU.
VIU v1.12.2 accidentally uploaded the asmdef files that causes some of the symbols not loaded.

**Comment by unoctium1** [comment]
@chengnay Ah sorry if I wasn't clear before, I'm specifically having this issue in projects not using Oculus Avatars, when VIU_OCULUSVR_AVATARS is undefined intentionally. 

I do have this fixed locally on my end so no worries at all if it won't make it into VIU until after the next update!

**Comment by chengnay** [comment]
@unoctium1 By default, VIU_OCULUSVR_AVATARS should be added to defined symbol field. If it wasn’t added, then there must be something wrong with VIU.

If you try using the batch file, will VIU_OCULUSVR_AVATARS symbol be added now?

**Comment by unoctium1** [comment]
After trying the batch file it seems that added in the symbol and then Quest control models from the Oculus Avatars folder showed up. 

I do have some projects where the Oculus avatars folder is not imported into the project, but other parts of the Oculus Integration are in use. In these cases, wouldn't it make sense for VIU to fallback to using the VIU Oculus Quest controller models?

**Comment by chengnay** [comment]
@unoctium1 Those projects don’t have Oculus avatars folder imported is because you manually remove it? How can I make my project import Oculus Integration without Oculus avatars folder? Just uncheck the folder when I import the SDK? In those cases, what render model did you see in your scene? Probably VIU didn’t consider such cases, I will double check.

**Comment by unoctium1** [comment]
@chengnay Yeah, you can either manually remove the folder or uncheck it when you import. In order to replicate you might also have to hit the 'repair define symbols' button in the preferences. In those cases, you don't see a render model at all. Button input does still work and VIU is correctly identifying the headset and controllers, it just isn't correctly assigning a model because [SetDeviceIndex method](https://github.com/ViveSoftware/ViveInputUtility-Unity/blob/51e992fdd17a0e671c3e8dca6b32439d09ebd618/Assets/HTC.UnityPlugin/ViveInputUtility/Scripts/Misc/OculusVRExtension/VIUOculusVRRenderModel.cs#L847-L889) does nothing due to the conditionals. 

I realize this is a bit of a niche issue, so I appreciate how much time you've spent helping out with this!

**Comment by unoctium1** [comment]
#220 does fix this by wrapping the Oculus-specific override for the RenderModelCreator in an #if block so that it falls back to the default RenderModelCreator and uses the VIU Quest controllers. I think that should be fine since it seems like VIUOculusVRRenderModel only does Avatar-specific things
